compiler/nim.nim

```diff
@@ -80,14 +80,25 @@ proc handleCmdLine(cache: IdentCache; conf: ConfigRef) =
     else:
       conf.projectPath = canonicalizePath(conf, getCurrentDir())
     loadConfigs(DefaultConfig, cache, conf) # load all config files
-    let scriptFile = conf.projectFull.changeFileExt("nims")
-    if fileExists(scriptFile):
-      runNimScript(cache, scriptFile, freshDefines=false, conf)
+
+    proc runNimScriptIfExists(scriptFile:string)=
+      if fileExists(scriptFile):
+        runNimScript(cache, scriptFile, freshDefines=false, conf)
+
+    # TODO:
+    # merge this complex logic with `loadConfigs`
+    # check whether these should be controlled via
+    # optSkipConfigFile, optSkipUserConfigFile
+    let config_nims = "config.nims"
+    runNimScriptIfExists(getSystemConfigPath(conf, config_nims))
+    runNimScriptIfExists(getUserConfigPath(config_nims))
+    runNimScriptIfExists(conf.projectPath / config_nims)
+    block:
+      let scriptFile = conf.projectFull.changeFileExt("nims")
+      runNimScriptIfExists(scriptFile)
       # 'nim foo.nims' means to just run the NimScript file and do nothing more:
-      if scriptFile == conf.projectFull: return
-    elif fileExists(conf.projectPath / "config.nims"):
-      # directory wide NimScript file
-      runNimScript(cache, conf.projectPath / "config.nims", freshDefines=false, conf)
+      if fileExists(scriptFile) and scriptFile == conf.projectFull: return
```

Comment:
Why does this have a ``fileExists`` check? If the same as ``conf.projectFull`` do return. And this should probably use ``cmpPaths``...

Reply:
> Why does this have a fileExists check? If the same as conf.projectFull do return

no, with your suggestion, `nim c foo.nims` would exit without error when `foo.nims` doesn't exist; with this PR (and in fact preexisting to this PR), it would give an error

> And this should probably use cmpPaths

done